File: content-solutions/recursion/rabbit-family.py
def solution(n):
    memo = {}

    def rabbits(m):
        if m <= 2:
            return 1
        if m in memo:
            return memo[m]
        memo[m] = rabbits(m - 1) + rabbits(m - 2)
        return memo[m]

    return rabbits(n)

File: content-solutions/recursion/rabbit-family.js
function solution(n) {
  const memo = new Map();
  function rabbits(m) {
    if (m <= 2) return 1;
    if (memo.has(m)) return memo.get(m);
    const value = rabbits(m - 1) + rabbits(m - 2);
    memo.set(m, value);
    return value;
  }
  return rabbits(n);
}
